test/react/functional-components/simple-10.js

```diff
@@ -0,0 +1,43 @@
+var React = require('react');
+// the JSX transform converts to React, so we need to add it back in
+this['React'] = React;
+
+let lazyVariable = null;
+
+function A(props) {
+  if (!lazyVariable) {
+    lazyVariable = 123;
+  }
+  return <div>Hello {lazyVariable}</div>;
+}
+
+function App() {
+  return (
+    <div>
+      <A />
+    </div>
+  );
+}
+
+function App2() {
+  return (
+    <div>
+      <A />
+    </div>
+  );
+}
+
+App.getTrials = function(renderer, Root) {
+  // keep a reference to the other root that also
+  // writes to the same variable
+  let x = App2;
```

Comment:
What does this do?

Reply:
Holds a reference to the function or it gets DCE.

Reply:
Can we do something like `App.App2 = App2` in the body instead?

Reply:
Sure, I'm not concerned about how we reference it. I'll change.